app/controllers/api/v2/green_lanes/goods_nomenclatures_controller.rb

```diff
@@ -4,12 +4,20 @@ module GreenLanes
       class GoodsNomenclaturesController < BaseController
         def show
           gn = ::GreenLanes::FetchGoodsNomenclatureService.new(params[:id]).call
-          possible_categorisations = ::GreenLanes::FindCategorisationsService.new.call(gn)
+          possible_categorisations = ::GreenLanes::FindCategorisationsService.call(goods_nomenclature: gn,
+                                                                                   origin: param_origin)
+
           presented_gn = GoodsNomenclaturePresenter.new(gn, possible_categorisations)
-          serializer = Api::V2::GreenLanes::GoodsNomenclatureSerializer.new(presented_gn, include: %w[applicable_measures possible_categorisations])
+          serializer = Api::V2::GreenLanes::GoodsNomenclatureSerializer.new(
+            presented_gn, include: %w[applicable_measures possible_categorisations]
+          )
 
           render json: serializer.serializable_hash
         end
+
+        def param_origin
+          params.fetch(:origin, nil)
```

Comment:
This doesn't handle a blank param, ie `?origin=`  - can this append `.presence`

Reply:
this is not outdated,  we are using `filter` now.